src/simulators/statevector/qubitvector_thrust.hpp

```diff
@@ -5056,17 +3919,116 @@ void QubitVectorThrust<data_t>::apply_pauli(const reg_t &qubits,
   add_y_phase(num_y, phase);
 
   if(x_mask == 0){
-    apply_function(multi_pauli_Z_func<data_t>(z_mask, phase),qubits);
+    apply_function(multi_pauli_Z_func<data_t>(z_mask, phase));
   }
   else{
-    auto qubits_sorted = x_qubits;
-    std::sort(qubits_sorted.begin(), qubits_sorted.end());
+    apply_function(multi_pauli_func<data_t>(x_mask, z_mask, x_max, phase) );
+  }
+}
+
+
+
+#ifdef AER_DEBUG
+
+template <typename data_t>
```

Comment:
For all the debugging messages would be great to use `spdlog`.

Reply:
I moved these messages using spdlog